include/nbl/builtin/hlsl/glsl_compat/core.hlsl

```diff
@@ -27,7 +27,17 @@ template<typename T>
 T atomicAdd(NBL_REF_ARG(T) ptr, T value)
 {
     // FUTURE: assert the scalart type is an integer type, then assert device traits maybe? but default to traits there?
-    return spirv::atomicIAdd<T>(ptr, spv::ScopeDevice, spv::MemorySemanticsMaskNone, value);
+    if (sizeof(T) == sizeof(uint32_t))
+        return spirv::atomicIAdd<T>(ptr, spv::ScopeDevice, spv::MemorySemanticsMaskNone, value);
+    return atomic64::atomicIAdd<T>(ptr, spv::ScopeDevice, spv::MemorySemanticsMaskNone, value);
+}
+template<typename T, typename Ptr_T> // DXC Workaround
+typename nbl::hlsl::enable_if<is_spirv_type_v<Ptr_T>, T>::type atomicAdd(Ptr_T ptr, T value)
```

Comment:
add this workaround overload to all atomics implemented here

Reply:
Done